### ISSUE TITLE
`Timer` appears to leak file descriptors on Mac

### ISSUE THREAD

**Issue by kwzhao** [DoumanAsh/async-timer#23]
The following program:

```rust
use std::time::{Duration, Instant};

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    let end = Instant::now() + Duration::from_secs(10);
    while Instant::now() < end {
        let timer = async_timer::new_timer(Duration::from_nanos(1000));
        timer.await;
    }
    Ok(())
}
```

produces this error on my Mac:

`Assertion 'fd != -1' failed. OSError(24): Too many open files`

Is this the correct behavior? I would have expected file descriptors to be released when the timer is consumed.

**Comment by kwzhao** [comment]
This error doesn't reproduce on Linux.

**Comment by DoumanAsh** [comment]
Pretty sure that should not happen
I will review code

**Comment by DoumanAsh** [comment]
Sorry it seems I missed implementing destructor for async timer
This problem is not visible on Linux because by default Linux systems has much bigger limit on file descriptors

**Comment by DoumanAsh** [comment]
I released 1.0.0-beta.15
Please check it

**Comment by kwzhao** [comment]
The fix works for me -- thanks!
